src/server.jl

```diff
@@ -48,36 +80,50 @@ function get_param(source::Vector{HTTP.Forms.Multipart}, name::String, required:
+function to_param_type(::Type{T}, strval::String; stylectx=nothing) where {T <: Number}
     parse(T, strval)
 end
 
-to_param_type(::Type{T}, val::T) where {T} = val
-to_param_type(::Type{T}, ::Nothing) where {T} = nothing
-to_param_type(::Type{String}, val::Vector{UInt8}) = String(copy(val))
-to_param_type(::Type{Vector{UInt8}}, val::String) = convert(Vector{UInt8}, copy(codeunits(val)))
-to_param_type(::Type{Vector{T}}, val::Vector{T}, _collection_format::Union{String,Nothing}) where {T} = val
+to_param_type(::Type{T}, val::T; stylectx=nothing) where {T} = val
+to_param_type(::Type{T}, ::Nothing; stylectx=nothing) where {T} = nothing
+to_param_type(::Type{String}, val::Vector{UInt8}; stylectx=nothing) = String(copy(val))
+to_param_type(::Type{Vector{UInt8}}, val::String; stylectx=nothing) = convert(Vector{UInt8}, copy(codeunits(val)))
+to_param_type(::Type{Vector{T}}, val::Vector{T}, _collection_format::Union{String,Nothing}; stylectx=nothing) where {T} = val
+to_param_type(::Type{Vector{T}}, json::Vector{Any}; stylectx=nothing) where {T} = [to_param_type(T, x; stylectx) for x in json]
+
+function to_param_type(::Type{Vector{T}}, json::Dict{String, Any}; stylectx=nothing) where {T}
+    if !isnothing(stylectx) && is_deep_explode(stylectx)
+        cvt = deep_object_to_array(json)
+        if isa(cvt, Vector)
+            return to_param_type(Vector{T}, cvt; stylectx)
+        end
+    end
+    error("Unable to convert $json to $(Vector{T})")
+end
 
-function to_param_type(::Type{T}, strval::String) where {T <: APIModel}
-    from_json(T, JSON.parse(strval))
+function to_param_type(::Type{T}, strval::String; stylectx=nothing) where {T <: APIModel}
+    from_json(T, JSON.parse(strval); stylectx)
 end
 
-function to_param_type(::Type{T}, json::Dict{String,Any}) where {T <: APIModel}
-    from_json(T, json)
+function  to_param_type(::Type{T}, json::Dict{String,Any}; stylectx=nothing) where {T <: APIModel}
+    from_json(T, json; stylectx)
 end
 
-function to_param_type(::Type{Vector{T}}, strval::String, delim::String) where {T}
+function to_param_type(::Type{Vector{T}}, strval::String, delim::String; stylectx=nothing) where {T}
     elems = string.(strip.(split(strval, delim)))
-    return map(x->to_param_type(T, x), elems)
+    return map(x->to_param_type(T, x; stylectx), elems)
 end
 
-function to_param_type(::Type{Vector{T}}, strval::String) where {T}
+function to_param_type(::Type{Vector{T}}, strval::String; stylectx=nothing) where {T}
     elems = JSON.parse(strval)
-    return map(x->to_param_type(T, x), elems)
+    return map(x->to_param_type(T, x; stylectx), elems)
 end
 
-function to_param(T, source::Dict, name::String; required::Bool=false, collection_format::Union{String,Nothing}=",", multipart::Bool=false, isfile::Bool=false)
+function to_param(T, source::Dict, name::String; required::Bool=false, collection_format::Union{String,Nothing}=",", multipart::Bool=false, isfile::Bool=false, style::String="form", is_explode::Bool=true)
```

Comment:
```suggestion
function to_param(T, source::Dict, name::String; required::Bool=false, collection_format::Union{String,Nothing}=",", multipart::Bool=false, isfile::Bool=false, style::String="form", location=:query, is_explode::Bool=default_param_explode(style))
```

Reply:
There was a mismatch here in the default value of `is_explode` compared to the client implementation. I guess that was inconsequential as of now, and using `default_param_explode()` should ensure to keep it uniform.